### PR TITLE
Add Verbose Logging to the Mobile App

### PR DESCRIPTION
**Pull request by jordanopos**
#### Summary of Changes (What does this PR do?)

- This PR adds more logging to the mobile app to help catch errors better.

#### Status of maturity (all need to be checked before merging):

- [x] I've tested this locally
- [x] I consider this code done
- [x] This change ready to hit production in its current state
- [x] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

## Summary by CodeRabbit

- **New Features**
  - Enhanced error handling across various components with detailed logging of exceptions and stack traces to improve debugging and support.

- **Bug Fixes**
  - Improved reliability by ensuring exceptions and errors are consistently logged throughout the app.

### CODE REVIEW

File: mobile/lib/screens/dashboard/dashboard_view.dart [hunk 1]
```diff
@@ -598,6 +598,7 @@ class _DashboardViewState extends State<DashboardView>
       },
       onError: (error) {
         debugPrint('error listening to location updates : $error');
+        logException(error, null);
```
Comment (Baalmart) [inline mobile/lib/screens/dashboard/dashboard_view.dart:601]: Hi @jordanopos do we need to "await" here too?

File: mobile/lib/blocs/insights/insights_bloc.dart [hunk 2]
```diff
@@ -42,7 +42,9 @@ class InsightsBloc extends Bloc<InsightsEvent, InsightsState> {
         forecastAirQuality: todayForecast.airQuality,
         forecastPm2_5: todayForecast.pm2_5,
       );
-    } catch (_) {}
+    } catch (exception, stackTrace) {
+      logException(exception, stackTrace);
```
Comment (Baalmart) [inline mobile/lib/blocs/insights/insights_bloc.dart:46]: @jordanopos do we need to "await" here too?

**Review by Baalmart — COMMENTED** [review]
Hi @jordanopos , thanks for this. I have just left a few comments for your review

File: mobile/lib/screens/dashboard/dashboard_view.dart (hunk 1 above)
Comment (jordanopos) [reply mobile/lib/screens/dashboard/dashboard_view.dart:601]: Hey @Baalmart , the function is designed to handle requests asynchronously, so we don't need to wait for it to finish before we can move on.

File: mobile/lib/blocs/insights/insights_bloc.dart (hunk 2 above)
Comment (jordanopos) [reply mobile/lib/blocs/insights/insights_bloc.dart:46]: Hey, @Baalmart , same thing here.

File: mobile/lib/screens/dashboard/dashboard_view.dart (hunk 1 above)
Comment (Baalmart) [reply mobile/lib/screens/dashboard/dashboard_view.dart:601]: thanks @jordanopos

File: mobile/lib/blocs/insights/insights_bloc.dart (hunk 2 above)
Comment (Baalmart) [reply mobile/lib/blocs/insights/insights_bloc.dart:46]: Alright @jordanopos

**Review by Baalmart — APPROVED** [review]
thanks @jordanopos

**Pull request merged**